patterns/1-initial/extensions-to-manage-contributions-at-scale.md

```diff
@@ -0,0 +1,81 @@
+
+## Context
+
+- More employees are contributing to strategic InnerSource codebases, some of which are scaling rapidly with new feature contributions. This is adding a huge code review backlog on a smaller group of maintainers, resulting in maintainer burnout as well as slowing down release of new capabilities.
+- Adding an excessive number of capabilities and code to repositories is making them difficult to maintain.
+- here is a growing backlog of new features and ideas for the products which have scaled.
+- The pattern applies in either scenario:
+   - Maintainers find themselves rejecting new feature ideas to narrow down the scope of the products. This is hampering innovation in the community and restricting further expansion.
+   - To reduce backlog, new features are getting released without thorough documentation, hardening, or testing, creating a poor user experience. This is also bloating the size of the codebases, adding a huge dependency graph and making them difficult to maintain.
+
+## Forces
+
+- Maintainers and product owners want to allow for expansion, encourage innovation and experimentation without being very restrictive on contributions, while also keeping good code and quality standards for user experience.
+- A large amount of time goes into hardening and thorough testing of features to meet product standards, but product owners may want to allow for faster release of new innovations for adopting products to explore before investing time in productionization of capabilities.
+- Maintainers want to encourage the community to share innovations that combine product capabilities with other use-cases without adding more dependencies to the primary repository.
+
+## Solutions
+
+Allowing extensions/plugins to high-scale InnerSource codebases or libraries can relieve the maintenance burden on repository maintainers and allow faster release of new features for adopting products to explore. This shifts maintenance of capabilities to extension owners and allows the primary repository to support capabilities that have been adopted more widely and are more strategic. Extensions provide a filter for new capabilities that may eventually move into the core of the project. Extensions also act as an incubation and community hardening environment, allowing for much of that hardening to happen organically rather than in a costly review process.
+
+In order for the extensions model to be successful, there are few architectural considerations to keep in mind:
+
+1. <b>Easy to create:</b> To obtain community participation, extensions need to be easy to create.
+   - Create Github templates for extensions to allow addition of new features in new repositories, retaining the same modular structure as the primary repository, with the framework to package and release extensions.
```

Comment:
```suggestion
   - Create a repository template that extensions should use as a starting point. This allows the extensions to add their new features in new repositories, separate from the core projects. The template should provide the same modular structure as the primary repository, and include the framework to package and release extensions.
```

I would not tie this to "GitHub" here, as we don't know what the adopters of the pattern are using.
I generalized it to "repository template" instead.

Also I would use singular i.e. "template".

Lastly I suggest to keep the sentences as short as possible.
I provided an example of that above.
While that means that overall there are more words in this case, each sentence is shorter and hopefully easier to understand.